GamePhysics: Move `defSymbols` to TermMap to avoid duplicates

jackwyand opened the pull request:
Contributes to #4036

After making this change I realized that it might only hurt with resolving #4126; maybe @balacij could take a look to see if this is a good change or not? I am not sure what is more of a priority at this point, having no duplicates within chunk tables or no duplicates across _all_ chunk tables.

JacquesCarette commented:
If the within table that are major issues are under control, that is indeed different!

JacquesCarette commented:
Can you give me an explicit list of the chunks that were conflicting? Was it the same chunks but add to two tables or something else?

jackwyand commented:
Sure!

In the `conceptChunk` list that is added to the ChunkDB, there is a list called `defSymbols` (Ln 149)
https://github.com/JacquesCarette/Drasil/blob/def8f55cbdd4cd2b2cd857e8b6958009b4513c06/code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Body.hs#L144-L153

Within this list, there are two other lists added, including `inputConstraints`:
https://github.com/JacquesCarette/Drasil/blob/def8f55cbdd4cd2b2cd857e8b6958009b4513c06/code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Unitals.hs#L26-L27

`forceCons`, added in `inputConstraints`:
https://github.com/JacquesCarette/Drasil/blob/def8f55cbdd4cd2b2cd857e8b6958009b4513c06/code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Unitals.hs#L317-L320

shares a UID with `force` since it uses that chunk in its creation
https://github.com/JacquesCarette/Drasil/blob/def8f55cbdd4cd2b2cd857e8b6958009b4513c06/code/drasil-example/gamephysics/lib/Drasil/GamePhysics/Unitals.hs#L334

`force` is then also added in `physicCon`, which contains all of the general physics concepts. `physicCon` is evidently also added in `conceptChunks`, causing the duplicate warning.

Hopefully this makes sense!

JacquesCarette commented:
Yes, it does.

I think the real problem here is that `forceCons` has the same `UID` as `force`. These are separate things. Unless `force` in GamePhysics is only ever "force of gravity"?

Feels to me that `constrained'` is a bad combinator.

balacij commented:
One key piece of information that @JacquesCarette mentioned about the "reasonable values" that the `UncertQ`s associate with `D-Q-D`s: "think about the order of magnitude of the numbers rather than the specific numbers." For example, in the context of a beam, a length of 50m is sane, a similar length 2 orders magnitude up or down is also fine. But a "beam" of length 1E-100m is not.
